### PR TITLE
Fixed type related errors

### PR DESCRIPTION
**Pull request by tisnik**
## Type of change

- [ ] Refactor
- [ ] New feature
- [x] Bug fix
- [ ] CVE fix
- [ ] Optimization
- [ ] Documentation Update
- [ ] Configuration Update
- [ ] Bump-up dependent library

## Description

There are some errors found during type checking (NPE can occurs etc.). So let's fix them.

## Related Tickets & Documents

- Related Issue [OLS-83](https://issues.redhat.com//browse/OLS-83)

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [x] Unit tests passed locally.
- [ ] If it is a core feature, I have added thorough tests.

## Testing
- N/A right now

### CODE REVIEW

File: tests/unit/utils/test_json_tools.py [hunk 1]
```diff
@@ -1,8 +1,10 @@
+from typing import Any
+
 import pytest
 
 from utils.json_tools import parse_and_check_json_markdown, parse_json_markdown
 
-correct_inputs = (
+correct_inputs: tuple[tuple[str, dict[str, Any]], ...] = (
```
Comment (onmete) [inline tests/unit/utils/test_json_tools.py:7]: Is it necessary to have typing in tests?

File: app/endpoints/ols.py [hunk 2]
```diff
@@ -98,9 +98,12 @@ def ols_request(llm_request: LLMRequest):
...
             llm_response.response = wrapper + "\n" + generated_yaml
-            config.conversation_cache.insert_or_append(
-                conversation, llm_request.query + "\n\n" + llm_response.response
-            )
+
+            if config.conversation_cache is not None:
+                config.conversation_cache.insert_or_append(
+                    conversation,
+                    llm_request.query + "\n\n" + str(llm_response.response or ""),
+                )
```
Comment (onmete) [inline app/endpoints/ols.py:106]: I might be missing some code context, but if this was changed by the tool, I'm not entirely sure if it's a good move. If `config.conversation_cache` is `None`, and it shouldn't be, we are missing a raise telling us that there is something wrong.

File: tests/unit/utils/test_json_tools.py (hunk 1 above)
Comment (tisnik) [reply tests/unit/utils/test_json_tools.py:7]: not strictly necessary

File: app/endpoints/ols.py (hunk 2 above)
Comment (tisnik) [reply app/endpoints/ols.py:106]: it is done by hand. Tooling can't do these types of changes right now.

Comment (tisnik) [reply app/endpoints/ols.py:106]: I inferred, that cache might not be configured and the service would still work (just w/o having conversation history).

File: tests/unit/utils/test_json_tools.py (hunk 1 above)
Comment (onmete) [reply tests/unit/utils/test_json_tools.py:7]: so if you remove it, would linting fail?

File: app/endpoints/ols.py (hunk 2 above)
Comment (onmete) [reply app/endpoints/ols.py:106]: makes sense

File: tests/unit/utils/test_json_tools.py (hunk 1 above)
Comment (tisnik) [reply tests/unit/utils/test_json_tools.py:7]: no, if we setup it correctly - can be done after we change project structure

Comment (tisnik) [reply tests/unit/utils/test_json_tools.py]: removed as agreed upon with @onmete

**Review by thoraxe — CHANGES_REQUESTED** [review]
Please squash and force push

**Pull request merged**